datafusion/src/physical_plan/csv.rs

```diff
@@ -263,12 +264,20 @@ impl CsvExec {
         filenames: &[String],
         options: &CsvReadOptions,
     ) -> Result<Schema> {
-        Ok(csv::infer_schema_from_files(
-            filenames,
-            options.delimiter,
-            Some(options.schema_infer_max_records),
-            options.has_header,
-        )?)
+        let mut schemas = Vec::new();
+        for filename in filenames {
```

Comment:
what's the reason for reimplementing `csv::infer_schema_from_files` here?

Reply:
Good sight! I deep into `csv::infer_schema_from_files` and find them nearly do the same thing.